src/main/java/com/exasol/adapter/request/renderer/PushdownSqlRenderer.java

```diff
@@ -0,0 +1,480 @@
+            final JsonObjectBuilder builder = JSON.createObjectBuilder();
+            final DataType.ExaDataType type = dataType.getExaDataType();
+            builder.add(TYPE, type.name().toUpperCase());
+            switch (type) {
+            case DECIMAL:
+                builder.add("precision", dataType.getPrecision());
+                builder.add("scale", dataType.getScale());
+                break;
+            case VARCHAR:
+            case CHAR:
+                builder.add("characterSet", dataType.getCharset().name().toLowerCase());
+                builder.add("size", dataType.getSize());
+                break;
+            case TIMESTAMP:
+                builder.add("withLocalTimeZone", dataType.isWithLocalTimezone());
+                break;
+            case INTERVAL:
+                builder.add("precision", dataType.getPrecision());
+                builder.add("fromTo", render(dataType.getIntervalType()));
+                if (dataType.getIntervalType().equals(DataType.IntervalType.DAY_TO_SECOND)) {
+                    builder.add("fraction", dataType.getIntervalFraction());
+                }
+                break;
+            case GEOMETRY:
+                builder.add("srid", dataType.getGeometrySrid());
+                break;
+            default:
+                break;
+            }
+            return builder.build();
+        }
+
+        private String render(final DataType.IntervalType intervalType) {
+            switch (intervalType) {
+            case DAY_TO_SECOND:
+                return "DAY TO SECONDS";
+            case YEAR_TO_MONTH:
+                return "YEAR TO MONTH";
+            default:
+                throw new IllegalStateException(ExaError.messageBuilder("F-VS-COM-JAVA-35")
+                        .message("Unimplemented interval type {{type}}.", intervalType).ticketMitigation().toString());
+            }
+        }
+
+        @Override
+        public JsonObject visit(final SqlFunctionScalarExtract function) throws AdapterException {
+            final JsonObjectBuilder builder = createObjectBuilderFor(function);
+            builder.add(NAME, "EXTRACT");
+            builder.add("toExtract", function.getToExtract());
+            builder.add(ARGUMENTS_KEY, convertListOfNodes(List.of(function.getArgument())));
+            return builder.build();
+        }
+
+        @Override
+        public JsonObject visit(final SqlFunctionScalarJsonValue function) throws AdapterException {
+            final JsonObjectBuilder builder = createObjectBuilderFor(function);
+            builder.add(NAME, function.getScalarFunction().name());
+            builder.add(ARGUMENTS_KEY, convertListOfNodes(function.getArguments()));
+            builder.add("returningDataType", render(function.getReturningDataType()));
+            builder.add("emptyBehavior", render(function.getEmptyBehavior()));
+            builder.add("errorBehavior", render(function.getErrorBehavior()));
```

Comment:
Maybe we should also use constants here and in the deserializer?

Reply:
done